packages/block-library/src/image/edit.js

```diff
@@ -209,6 +211,7 @@ export function ImageEdit( {
 				allowedTypes: ALLOWED_MEDIA_TYPES,
 				onError: ( message ) => {
 					noticeOperations.createErrorNotice( message );
+					setState( { hasError: true } );
```

Comment:
How would you feel using the following

```
setAttributes( {
  src: undefined, 
  id: undefined,
  url: undefined,
} );
```

instead of `setState` ?

I got inspired from the `audio` block which apparently works as expected 
https://github.com/WordPress/gutenberg/blob/a610923b7728f1192f9745a55390a7abb0df576e/packages/block-library/src/audio/edit.js#L61

In fact, in my tests by using only they above code change to the whole file is enough!
![SS 2020-08-31 at 18 59 06](https://user-images.githubusercontent.com/12430020/91741454-6986a700-ebbd-11ea-9756-406d73466043.gif)

Reply:
I'm a bit late to the discussion 😄